test/spec/vis/layers-factory.spec.js

```diff
@@ -145,6 +150,50 @@ describe('vis/layers-factory', function () {
         });
       });
     });
+
+    describe('Is high resolution screen', function () {
+      beforeEach(function () {
+        spyOn(LayersFactory, 'isRetina').and.returnValue(true);
+      });
+
+      describe('urlTemplateRetina is not defined', function () {
+        it("should not convert to '" + urlTemplateRetina + "'", function () {
+          var layerModel = layersFactory.createLayer('tiled', {
+            urlTemplate: urlTemplate
+          });
+
+          expect(layerModel.get('urlTemplate')).toEqual(urlTemplate);
+        });
+      });
+
+      describe('urlTemplateRetina is defined', function () {
+        it("should convert to '" + urlTemplateRetina + "'", function () {
+          var layerModel = layersFactory.createLayer('tiled', {
+            urlTemplate: urlTemplate,
+            urlTemplateRetina: urlTemplateRetina
+          });
+
+          expect(layerModel.get('urlTemplate')).toEqual(urlTemplateRetina);
+        });
+      });
+    });
+
+    describe('Is not high resolution screen', function () {
+      beforeEach(function () {
+        spyOn(LayersFactory, 'isRetina').and.returnValue(false);
+      });
+
+      describe('urlTemplateRetina is defined', function () {
+        it("should not convert to '" + urlTemplateRetina + "'", function () {
```

Comment:
`it('should use the default urlTemplate when the urlTemplateRetina is defined');`

Reply:
👍